include/nil/blueprint/manifest.hpp

```diff
@@ -156,25 +158,7 @@ namespace nil {
                 return t < lt.t;
             }
 
-            manifest_constant_type intersect(manifest_constant_type lt) const {
-                if (t == manifest_constant_type::type::UNSAT ||
-                    lt == manifest_constant_type::type::UNSAT) {
-
-                    return manifest_constant_type::type::UNSAT;
-                } else if (t == manifest_constant_type::type::NONE &&
-                           lt == manifest_constant_type::type::REQUIRED ||
-                           t == manifest_constant_type::type::REQUIRED &&
-                           lt == manifest_constant_type::type::NONE) {
-                    return manifest_constant_type::type::UNSAT;
-                } else if (t == manifest_constant_type::type::REQUIRED ||
-                           lt == manifest_constant_type::type::REQUIRED) {
-                    return manifest_constant_type::type::REQUIRED;
-                } else if (t == manifest_constant_type::type::NONE ||
-                           lt == manifest_constant_type::type::NONE) {
-                    return manifest_constant_type::type::NONE;
-                }
-                return manifest_constant_type::type::UNSAT;
-            }
+            manifest_constant_type intersect_with_compiler(const compiler_manifest &manifest) const;
```

Comment:
ok for me, just not exactly clear why need to introduce new `intersect_with_compiler` instead `intersect`. Different signature for different parameters a little bit confused

Reply:
Yeah, you are right, going to rename it back to `intersect`.